safekeeper/src/timeline.rs

```diff
@@ -679,6 +696,9 @@ impl Timeline {
             wal_backup::delete_timeline(&self.ttid).await?;
         }
         let dir_existed = delete_dir(&self.timeline_dir).await?;
+        if shared_state.sk.is_evicted() {
```

Comment:
It might be possible that this line would run several times, in case someone retries deletion request?

Reply:
In theory yes. That's why I added this as the last step, to avoid multiple decrement on retries. This is of course still not bullet proof, but should be ok nearly always, and it is ok for metric to be approximate.